api/rs/build/lib.rs

```diff
@@ -273,6 +273,14 @@ pub fn compile(path: impl AsRef<std::path::Path>) -> Result<(), CompileError> {
 }
 
 /// Same as [`compile`], but allow to specify a configuration.
+///
+/// Compile `ui/hello.slint` and select the "material" style:
+/// ```rust,no_run
+/// let config =
+///     slint_build::CompilerConfiguration::new()
+///     .with_style("material".into());
+/// slint_build::compile_with_config("ui/hello.slint", config).unwrap();
```

Comment:
Unrelated to this PR, but i wonder if we should not provide a `compile`  function directly in the CompilerConfiguration,  so this match the builder patern in which you call `build()` at the end

```rust
slint_build::CompilerConfiguration::new()
     .with_style("material".into())
     .compile("ui/hello.slint")
     .unwrap();
```

Reply:
Yeah. I guess then it would be a `slint_build::Compiler`? Seems odd to call `compile` on a `configuration` :). Details though, I agree about the current pattern being a bit weird (my fault!).